File: st-util/src/test/java/de/saring/util/unitcalc/CalculationUtilsTest.java
package de.saring.util.unitcalc;

import org.junit.Test;
import static org.junit.Assert.*;

/**
 * This class contains all unit tests for the CalculationUtils class.
 *
 * @author Stefan Saring
 */
public class CalculationUtilsTest {
    
    /**
     * Tests the appropriate method.
     */
    @Test
    public void testCalculateDistance () {
        assertEquals (0f, CalculationUtils.calculateDistance (0, 0), 0f);
        assertEquals (20f, CalculationUtils.calculateDistance (20, 3600), 0.01f);
        assertEquals (49.165054f, CalculationUtils.calculateDistance ((float) 35.3, 5014), 0.0001f);
    }
    
    /**
     * Tests the appropriate method.
     */
    @Test
    public void testCalculateAvgSpeed () {
        assertEquals (Float.NaN, CalculationUtils.calculateAvgSpeed (0, 0), 0f);
        assertEquals (20f, CalculationUtils.calculateAvgSpeed (20, 3600), 0.01f);
        assertEquals (35.3f, CalculationUtils.calculateAvgSpeed ((float) 49.165054, 5014), 0.00001f);
    }
    
    /**
     * Tests the appropriate method.
     */
    @Test
    public void testCalculateDuration () {
        assertEquals (0, CalculationUtils.calculateDuration (0, 0));
        assertEquals (3600, CalculationUtils.calculateDuration (20, 20));
        assertEquals (5017, CalculationUtils.calculateDuration ((float) 49.2, (float) 35.3));
    }
}

File: st-util/src/test/kotlin/de/saring/util/unitcalc/CalculationUtilsTest.kt
package de.saring.util.unitcalc

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

/**
 * This class contains all unit tests for the CalculationUtils class.
 *
 * @author Stefan Saring
 */
class CalculationUtilsTest {

    /**
     * Tests the appropriate method.
     */
    @Test
    fun testCalculateDistance() {
        assertEquals(0.0, CalculationUtils.calculateDistance(0.0, 0))
        assertEquals(20.0, CalculationUtils.calculateDistance(20.0, 3600), 0.01)
        assertEquals(49.165054, CalculationUtils.calculateDistance(35.3, 5014), 0.0001)
    }

    /**
     * Tests the appropriate method.
     */
    @Test
    fun testCalculateAvgSpeed() {
        assertEquals(java.lang.Double.NaN, CalculationUtils.calculateAvgSpeed(0.0, 0))
        assertEquals(20.0, CalculationUtils.calculateAvgSpeed(20.0, 3600), 0.01)
        assertEquals(35.3, CalculationUtils.calculateAvgSpeed(49.165054, 5014), 0.00001)
    }

    /**
     * Tests the appropriate method.
     */
    @Test
    fun testCalculateDuration() {
        assertEquals(0, CalculationUtils.calculateDuration(0.0, 0.0))
        assertEquals(3600, CalculationUtils.calculateDuration(20.0, 20.0))
        assertEquals(5017, CalculationUtils.calculateDuration(49.2, 35.3))
    }
}
